### ISSUE TITLE
combined toolbar and titlebar

### ISSUE THREAD

**Issue by brainchild0** [linuxmint/nemo#2338]
GNOME applications, such as Nautilus, derive an optimized use of window space by combining the titlebar, menubar, and toolbar of each application into a single customized titlebar.

Cinnamon applications, such as Nemo, in contrast, revert to a traditional, plain separation of these components. 

While this choice may appeal to the aesthetics of some, it causes a greater portion of the space to be consumed by administrative features rather than the main application pane. The combination used by GNOME applications is increasingly favored for its simplicity and optimization.

Would returning to some variation of the combined approach be considered for the core Cinnamon applications?

**Comment by JosephMcc** [comment]
We take this approach on some Mint apps and where we think it makes sense but we won't be doing it just because. The traditional style of Cinnamon is part of it's appeal to it's users. Thanks for the feedback and input.

**Comment by brainchild0** [comment]
I understand that a classical appearance is a design objective of Cinnamon, as the project is one largely defined by its opposition to the strange direction taken by GNOME desktop. Yet the widespread abandonment of GNOME by much of its user base creates an opening for a moderate level of innovation to appear in Cinnamon, as many of the users who left would support these features as long as Cinnamon does not for some reason follow GNOME along the path of doom.

Put simply, the proposed design might not be a serious challenge to the reasons why users find Cinnamon appealing.

Perhaps the compacted layout could be optional, if dislike by certain users is a concern.

The current suggestion is not *just because*, but rather directed toward a more efficient use of space. Title bar, menu bar, and toolbar collectively consume considerable real estate, which may amount to a considerable portion of the overall window height, especially in scenarios involving many small windows.

I would ask you to be open to the idea in the future, even if it is not a current priority.

**Comment by salvoc81** [comment]
I also believe it would be nice addition, and I think it could be an option in the settings.
This would help keeping the Cinnamon aesthetic unchanged, and at the same time save some screen space for users who might need it.

**Comment by brainchild0** [comment]
> I think it could be an option in the settings.

Yes, I think   these kinds of options often have been  found in per application settings, but a helpful solution might be the introduction of a global setting.

I wish this issue would remain open, to help it gain notice by others who  may wish to contribute their own views about which directions they would like to see in the development of the system.

**Comment by JosephMcc** [comment]
It was closed because the original issue/question/request was answered. And because the entire idea of making it a setting, especially a global setting, is unrealistic and not really doable.

**Comment by LB22** [comment]
I am currently on a small laptop screen and I would **_really_** like the option to have a more space for my open windows.

I just switched from Windows 10 to Linux Mint Cinnamon and immediately felt awkward with the top bar "[Title] - [Application]" taking up what I think is unnecessary space. I first noticed it while starting browsing in Firefox. Just my two cents.

**Comment by Jeremy7701** [comment]
Even on a large screen its a waste of space.

**Comment by brainchild0** [comment]
@LB22: Firefox may be customized for a hidden title bar (through the toolbar customization system) the same as when running under Cinnamon versus any other desktop system. In fact, any application with such a feature works fine under Cinnamon. Presently, however, the Cinnamon applications themselves have no support for the feature. Many Cinnamon applications are based on GNOME applications, which all rely heavily on client-side decorations, but the Cinnamon variations have this design element removed.

The current request is for including similar support, for optional  use,  in Cinnamon  core applications.  You may configure Firefox on your Cinnamon system as you would like.

**Comment by LB22** [comment]
@brainchild0 Thank you for pointing me in the right direction in regards to configuring Firefox, I managed to get rid of the title bar. I also hid the menubar inside Nemo which freed up some space too. Still, combining the title bar and toolbar might not be a bad idea, but I'm going to be happy enough with the customization I made and withdraw from this discussion and leave this up to you. Thanks again! :+1:
